src/nett/nett.py

```diff
@@ -255,30 +256,30 @@ def analyze(config: str,
 
     def _execute_job(self, job: Job) -> Future:
         brain: "nett.Brain" = deepcopy(self.brain)
-
-        # for train
-        if job.estimate_memory or job.mode in ["train", "full"]: # TODO: Create a memory estimate method for test
-            try:
-                # initialize environment with necessary arguments
-                with self._wrap_env("train", job.port, job.env_kwargs()) as train_environment:
-                    brain.train(train_environment, job)
-            except Exception as e:
-                self.logger.exception(f"Error in training: {e}")
-                raise e
-
-        # for test
-        if job.mode in ["test", "full"]:
-            try:
-                # initialize environment with necessary arguments
-                with self._wrap_env("test", job.port, job.env_kwargs()) as test_environment:
-                    brain.test(test_environment, job)
-
-                    # if job.estimate_memory:                 
-                    #     with open(job.paths["base"] / "mem.txt", "w") as file:
-                    #         file.write(str(nvmlDeviceGetMemoryInfo(nvmlDeviceGetHandleByIndex(job.device)).used))
-            except Exception as e:
-                self.logger.exception(f"Error in testing: {e}")
-                raise e
+        
+        if job.estimate_memory: # estimate memory uses train env for estimation
+            modes = ["train"]
+        elif job.mode == "full":
+            modes = ["train", "test"]
+        else: # test or train
+            modes = [job.mode]
+
+        # loop over modes to validate then run the environment
```

Comment:
@Zach-Attach I remember you had mentioned in one of our meetings that the validation checks run more than once, could this be why?